surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,360 @@
+
+### Do you often use a debugger with Rust?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> Knowing how often a debugger is used with Rust is valuable, but need to know
+> whether that is because of the quality of the experience or because it
+> is/isn't often needed.
+
+### How do you start a debugger (when debugging Rust)?
+Type: select all that apply
+
+- Core dumps
+- Attaching to a process
+- Launching executable
+- From an IDE
+
+> *Justification:*
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
+
+### What debuggers have you heard of?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> If users aren't aware of certain types of debugger than raising awareness or
+> making it easier to start these debuggers might be valuable.
+
+### What debuggers do you use with Rust?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> Debuggers that are actually used is useful.
+
+### Which debugger interfaces do you use with Rust?
+Type: select all that apply
+
+- Terminal interface
+- Visual Studio Code for `gdb`
+- Visual Studio Code for `lldb`
+- `gdb`'s TUI mode
+- Vim/Neovim's `termdebug` for `gdb`
+- Pernosco for `rr`
+- Visual Studio
+- XCode
+
+> *Justification:*
+>
+> Interface used for a debugger materially changes the experience of using a
+> debugger and the advantages/disadvantages of it.
+
+### What pain points have you experienced using a debugger with Rust?
+Type: select all that apply
+
+- Often step too many times and need to restart from scratch
+- Don't have a source view of the current line being run
+- Unintuitive interface/errors
+- Expression parser doesn't support expressions that you want to write
+- Optimized-out code
+- Conditional breakpoints don't trigger
+- Difficult to trace dataflow
+- Too much detail/irrelevant information (i.e. assembly views)
+- Incorrect information displayed (i.e. line numbers, variable values)
+- Current line number jumps around unexpectedly
+
+> *Justification:*
+>
+> It might be possible for us to address these - or they might be the result of
+> the compiler's debuginfo output.
+
+
+### Does your Rust debugging typically cross language boundaries?
+Type: yes/no
+
+> *Justification:*
+>
+> If lots of users are using debuggers when doing cross-language interop, then
+> support for this could be improved.
+
+### Which programming language(s) do you debug across (with Rust)?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- ...
+
+> *Justification:*
+>
+> Debugging Rust + C is very different from Rust + Python and the quality of
+> the experience is likely very different too.
+
+
+### Have you ever written a visualization for a debugger (with Rust)?
+Type: select all that apply
```

Comment:
This question should be explicitly marked as optional

Reply:
Fixed!